parsers/thueringen.py

```diff
@@ -0,0 +1,103 @@
+from urllib.request import urlopen
+from bs4 import BeautifulSoup as parse
+import re
+from utils import Parser
+from datetime import datetime, timedelta
+from pyopenmensa.feed import LazyBuilder
+
+calendar_week_regex = re.compile('\d{4}-\d{1,2}$')
+week_start_end_date_regex = re.compile('\d{1,2}\.\d{1,2}\.\d{4}')
+fees_regex = re.compile('Bedienstete.*')
+amount_regex = re.compile('\d{1,2},\d{1,2}')
+days_regex = re.compile('day_\d$')
+ingredients_regex = re.compile('Inhalt:.*')
+
+
+def parse_url(url, today=False):
+	canteen = LazyBuilder()
+	content = urlopen(url).read()
+	document = parse(content, 'lxml')
+	employees_fee = 1.8
+	guests_fee = 3.2
+	fees = document.find_all('p', {'class': 'MsoNormal'})
+	for fee_candidate in fees:
+		fee_string = fees_regex.findall(fee_candidate.text)
+		for s in fee_string:
+			amount_strings = amount_regex.findall(s)
+			if len(amount_strings) is not 2:
+				continue
+			employees_fee = float(amount_strings[0].replace(',', '.'))
+			guests_fee = float(amount_strings[1].replace(',', '.'))
+	options = document.find_all('option', value=calendar_week_regex)
+	for option in options:
+		try:
+			_ = option['selected']
+			start_end_dates = week_start_end_date_regex.findall(option.text)
+			start_date_str = start_end_dates[0]
+			mensa_start_date = datetime.strptime(start_date_str, '%d.%m.%Y')
+			days = document.find_all('div', id=days_regex)
+			days_open = []
+			for day in days:
+				day_id = int(day['id'][-1:])
+				days_open.append(day_id)
+				current_date = mensa_start_date + timedelta(days=day_id-2)
+				meals_data = day.find_all('tr')
+				for meal_data in meals_data:
+					meals = meal_data.find_all('td')
```

Comment:
Same here: Extract a function. Small functions are awesome, because you can quickly understand them, whereas such a big function is difficult to read.

Anywhere you _can_ reasonably extract a function, do it. :)

Reply:
I extracted and split the parsing into 3 different functions